django/contrib/postgres/constraints.py

```diff
@@ -0,0 +1,90 @@
+from django.db.models import CheckConstraint, F
+from django.db.models.sql import Query
+
+__all__ = ['ExcludeConstraint']
+
+
+class ExcludeConstraint(CheckConstraint):
+    template = 'CONSTRAINT %(name)s EXCLUDE USING %(index_type)s(%(expressions)s)%(where)s'
+
+    def __init__(self, *, expressions=(), index_type='GIST', operators=(), check=None, name=''):
+        if not len(expressions) == len(operators):
+            raise ValueError('Expressions and operators need to be of the same length.')
```

Comment:
This makes we wonder if passing `expressions` as a list of 2-element tuples would be more appropriate. I guess the current break down of `expressions` and `operators` is more coherent with the actual expressions interface and the exclusion constraint syntax.